docs/guide/SecurityBC.md

```diff
@@ -273,3 +273,234 @@ Here is the response that comes back:
+
+Oathkeeper exposes two ports as two services. One service is the API service, and the other is the Proxy service. Long-term, we’ll be using the API service, which will be queried by the next generation API Gateway using the Access Control Decision API  (see [REST API | ORY Oathkeeper](https://www.ory.sh/oathkeeper/docs/reference/api/#access-control-decision-api) ) that Oathkeeper provides, but for now we will be using the Proxy service, and exposing that via Ingress. External URLs to services protected by Oathkeeper will be pointed at  the Oathkeeper Proxy ingress, which will then proxy access to the internal services at those URLs and apply access control rules.
+
+Oathkeeper will be configured to generate and sign a JWT containing claims that internal services can check and verify by pointing at the JWKS that Oathkeeper will publish (which is part of the configuration) at the well-known URL for JWKS on the API service (see [REST API | ORY Oathkeeper](https://www.ory.sh/oathkeeper/docs/reference/api/#lists-cryptographic-keys) ). If a service does this, then it is operating under a basic zero trust regime, as it will not be possible to call that service except with a token that has been generated by Oathkeeper, and Oathkeeper will only generate a token if the access rules for the given URL allow access.
+
+
+### Debugging
+The responses and logs from Oathkeeper tend to be pretty informative, definitely start there. Make sure you’re looking at the logs for the request that matters. Oathkeeper will also be logging health checks and similar. 
+
+Some possible debugging actions that I’ve found useful in different circumstances:
+
+* make a match much, much more permissive (replacing the entire beginning with `<.*>` and then having the minimum currently unique suffix is good)
+* double check that each internal URL is the proper internal URL by checking it is accessible inside the cluster with curl
+* look at the IdP logs
+* make sure the domains and ports for the introspection and the external token endpoint are identical. Keycloak at least doesn’t like it if they are not
+* point the Oathkeeper rule at https://httpbin.org/ , usually the `/anything` path prefix which will reflect back everything it gets, making it easy to see what the service will see
+
+### In Addition to Helm Chart
+The following piece will be needed in addition to the helm chart.
+
+#### JWKS Secret
+
+A secret should be created with key `mutator.id_token.jwks.json` and value of a JWKS suitable for use with Oathkeeper. An initial one can be generated as described at [Configure and Deploy | ORY Oathkeeper](https://www.ory.sh/oathkeeper/docs/configure-deploy#cryptographic-keys). It will contain both public and private keys.

```

Comment:
```suggestion
A secret should be created with key `mutator.id_token.jwks.json` and the value of a JWKS suitable for use with Oathkeeper. An initial one can be generated as described in [Configure and Deploy | ORY Oathkeeper](https://www.ory.sh/oathkeeper/docs/configure-deploy#cryptographic-keys). It will contain both public and private keys.

```